### PR TITLE
[Issue 10695] Fix Button Text Cut Off on 'Orders' Screen

### LINKED ISSUE #10695: 'Troubleshooting' Button Text Cut Off on 'Orders' Screen
**Describe the bug**
I've encountered a layout issue on the 'Orders' screen. The 'Troubleshooting' button's text is cut off, which seems to be a UI design flaw where the button isn't allocated enough space to display its full text. This problem occurs at the top of the screen, where there is an error message indicating that the app couldn't load user data.

**To Reproduce**
1. Go to 'Orders' screen.
2. Simulate an error. 
3. Observe the top section of the screen where the error message "We couldn't load your data..." is displayed.

**Screenshots**

<kbd></kbd>

**Expected behavior**
The 'Troubleshooting' button should be fully visible to users, allowing them to easily identify and click on it for further assistance.

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Mobile Environment**

Please include:
- Device: Samsung Galaxy S24
- Android version: Android 14
- WooCommerce Android version: 17.1

**WordPress Environment**
<details>
```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin.
```
</details>

### PR DESCRIPTION
**Pull request by jd-alexander**
Closes: #10695

### Description

This PR addresses the layout issue on the 'Orders' screen where the 'Troubleshooting' button text was previously cut off. The changes involve updating the layout constraints in the XML to ensure the button is fully visible and correctly positioned relative to the surrounding UI elements.

### Testing instructions

1. Go to 'Orders' screen.
2. Simulate an error. You can use this patch. 
[shows_the_notice_card.patch](https://github.com/woocommerce/woocommerce-android/files/14188379/shows_the_notice_card.patch)
3. Observe the top section of the screen where the error message "We couldn't load your data..." is displayed.
4. Notice that the Troubleshooting word isn't cut off.

### Images/gif

<kbd></kbd>

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Review by kidinov — COMMENTED** [review]
Thanks for working on this! @jd-alexander 

I think that sort of fixes the issue on some devices, but still, on small screens the layout doesn't look great as the buttons are not constraint to each other and nothing stops them to overlay on each other

image

Shell we constraint them to each other (and maybe to space in between to keep min distance in between?) so on narrow devices they the text will be 2 lines. That not ideal but probably better than what we have now

What do you think about reworking the layout overall? It seems like there is a constraint layout inside another constraint layout, which is redundant and could negatively impact performance.

**Comment by jd-alexander** [comment]
>What do you think about reworking the layout overall? It seems like there is a constraint layout inside another constraint layout, which is redundant and could negatively impact performance. 

Indeed. I am going to rework the layout and optimize it for all screen sizes! Thanks for the review :)

File: WooCommerce/src/main/res/layout/view_expandable_notice_card.xml [hunk 1]
```diff
@@ -24,58 +24,47 @@
...
+            android:layout_marginStart="@dimen/minor_00"
+            android:layout_marginEnd="8dp"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toStartOf="@+id/btn_secondaryAction"
+            app:layout_constraintStart_toEndOf="@+id/guideline7"
+            app:layout_constraintTop_toBottomOf="@+id/notice_message"
+            tools:text="@string/error_troubleshooting" />
 
         <androidx.constraintlayout.widget.Guideline
             android:id="@+id/guideline7"
```
Comment (kidinov) [inline WooCommerce/src/main/res/layout/view_expandable_notice_card.xml]: np: `guideline7` probably not the best naming

**Review by kidinov — APPROVED** [review]
I left one np comment, other than that LGTM!

File: WooCommerce/src/main/res/layout/view_expandable_notice_card.xml (hunk 1 above)
Comment (jd-alexander) [reply WooCommerce/src/main/res/layout/view_expandable_notice_card.xml]: Agreed 😅 I will change it.

**Comment by jd-alexander** [comment]
Hi @kidinov the additional constraint layout that is in the layout is needed for animation reasons so it is put back within the layout. 

Below are screenshots from the actual layout for verification. All good to go just wanted another 👍🏾  from you since I made a change you didn't approve initially. Thanks 

Screenshot 2024-02-12 at 7 47 25 PM
Screenshot 2024-02-12 at 7 47 05 PM

**Comment by kidinov** [comment]
@jd-alexander I think you still can avoid having nested layouts by using a group

```
<androidx.constraintlayout.widget.Group
            android:id="@+id/noticeMorePanel"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            app:constraint_referenced_ids="notice_message,btn_mainAction,btn_secondaryAction" />
```

**Comment by jd-alexander** [comment]
> @jd-alexander I think you still can avoid having nested layouts by using a group
> 
> ```
> <androidx.constraintlayout.widget.Group
>             android:id="@+id/noticeMorePanel"
>             android:layout_width="wrap_content"
>             android:layout_height="wrap_content"
>             app:constraint_referenced_ids="notice_message,btn_mainAction,btn_secondaryAction" />
> ```

Cool 👍🏾

**Comment by jd-alexander** [comment]
Hi @kidinov I implemented the changes. Just to be sure we are on the same page, this is my understanding: 

1. Remove the nested `ConstraintLayout` and utilize a `Group` to manage the visibility of related views. 
2. The `ToggleButton`, `MaterialTextView`, and both `MaterialButton` elements are now directly placed within the main `ConstraintLayout`. 
3. The `Group` with the ID `noticeMorePanel` references the `notice_message`, `btn_mainAction`, and `btn_secondaryAction` views, allowing for collective visibility control without the need for additional nested layouts. 

Performing the above accomplishes the goal you stated of optimizing the layout hierarchy which improves performance and maintainability.

**Review by kidinov — APPROVED** [review]
I see that before `ConstraintLayout` had `android:visibility="gone"` prop set, but at the same time it looks like that we "expand" it via code anyway by default if it's visible, so it could be redundant before. But I simulate visibility of this card via the code so maybe I am missing something. If in your testing it was expanded by default the please go ahead and merge

Thanks!

**Comment by jd-alexander** [comment]
>I see that before ConstraintLayout had android:visibility="gone" prop set, but at the same time it looks like that we "expand" it via code anyway by default if it's visible, so it could be redundant before. But I simulate visibility of this card via the code so maybe I am missing something. If in your testing it was expanded by default the please go ahead and merge

> Thanks!

Yeah it was expanded by default for me and the value does seem redundant. If I notice any irregularities I will fix it again. Thanks for sharing this!

**Pull request merged**